esp-hal/src/uart/mod.rs

```diff
@@ -1177,8 +1178,16 @@ impl<'d> UartRx<'d, Async> {
             // No point in listening for timeouts, as we're waiting for an exact amount of
             // data. On ESP32 and S2, the timeout interrupt can't be cleared unless the FIFO
             // is empty, so listening could cause an infinite loop here.
-            self.wait_for_buffered_data(buf.len(), buf.len(), false)
-                .await?;
+            //
+            // The threshold is set to 75% of the FIFO to reduce the likelihood of overflows.
```

Comment:
We should be using rx configs `fifo_full_threshold` here instead of hard coding this IMO. The default is 120 (lifted from idf's defaults), which gives an 8 byte window to drain the fifo before filling up.